api/openapi.yaml

```diff
@@ -1685,34 +1661,81 @@ components:
-              enum:
-                - Aborted
-                - TimedOut
-                - OneOrMoreAssetFailedToScan
-                - DiscoveryFailed
-                - Unexpected
-                - NothingToScan
-                - Success
+            status:
+              $ref: '#/components/schemas/ScanStatus'
             summary:
               $ref: '#/components/schemas/ScanSummary'
         - $ref: '#/components/schemas/ScanTemplate'
 
+    ScanStatus:
+      type: object
+      required:
+        - state
+        - reason
+        - lastTransitionTime
+      properties:
+        state:
+          description: |
+            Describes the state of Scan.
+            
+            | State      | Description                                       |
+            | ---------- | ------------------------------------------------- |
+            | Pending    | Scan is waiting to be scheduled                   |
+            | Discovered | Assets have been found for the Scan               |
+            | InProgress | The AssetScans are running on the target machines |
+            | Aborted    | Scan has been aborted through the API             |
+            | Failed     | At least one AssetScan ended with an error        |
+            | Done       | Every AssetScan ended successfully                |
+          type: string
+          enum:
+            - Pending
+            - Discovered
+            - InProgress
+            - Aborted
+            - Failed
+            - Done
+        reason:
+          description: |
+            Machine readable reason for state transition.
+            
+            | State      | Reason            | Description                                                |
+            | ---------- | ----------------- | ---------------------------------------------------------- |
+            | Pending    | Created           | Scan is pending as it has been newly created               |
+            | Discovered | AssetsDiscovered  | Assets that fall into the scope of the Scan are discovered |
+            | InProgress | AssetScansRunning | AssetScans are currently running                           |
+            | Aborted    | Cancellation      | Scan has been aborted due to cancellation request          |
+            | Failed     | Cancellation      | Scan has been aborted due to cancellation request          |
+            | Failed     | AssetScanFailed   | At least one AssetScan has ended with an error             |
+            | Failed     | Timeout           | Scan has been timed out                                    |
+            | Done       | NothingToScan     | There are no assets that need to be scanned                |
+            | Done       | Success           | Scan ended successfully without any errors                 |
+          type: string
+          enum:
+            - Created
+            - AssetsDiscovered
+            - AssetScansRunning
+            - Cancellation
```

Comment:
I suggest `Cancelled` here to make these words a bit more consistent. Maybe also `Timed out` and `Succeeded`?

Reply:
> the scan state is aborted due to cancellation

following this format and given that we kept the `cancellation` reason, should we then change `Created` to `Creation`?

Reply:
Just chiming in with a quick question: how come `Cancellation` can result in both `Aborted` and `Failed` states?

Reply:
I answered under Ramiz's comment: https://github.com/openclarity/vmclarity/pull/1048#discussion_r1445152125